src/a-runtime-console/kubernetes/model/pipelinestage.model.ts

```diff
@@ -43,10 +43,9 @@ export class PipelineStage {
             this.environmentName = name;
             let urlMap = se.serviceUrls;
             this.serviceUrlMap = urlMap;
-
-            let buildConfigName = build.buildConfigName;
-            if (buildConfigName) {
-              this.serviceUrl = urlMap[buildConfigName] || '';
+            let servUrlName: string[] = Object.keys(urlMap);
+            if (servUrlName && servUrlName[0]) {
+              this.serviceUrl = urlMap[servUrlName[0]] || '';
```

Comment:
The assumption on the 0 indexed item being the correct service url feels a little dangerous, but after analyzing the code it's not like the rest of it is capable of handling multiple service url's at this time. It would be nice to have a unit test that fails before this change and passes afterwards; you can log some of the data when connected to production so it's not too hard to write.

Reply:
@jiekang yes as per my understanding earlier I mean in the legacy wizard as you are aware we had `app_name == repo_name == artifact_id` and here `build.buildConfigName == app_name` and `urlMap: { artifact_id : 'route_stage'}` it used to work as both buildConfigName and key name under urlMap used to be same.
> this.serviceUrl = urlMap[buildConfigName] || '';

now with new launcher flow as these values are not same so I fetched `Object.keys(urlMap)` and as far I've seen it always has just one key and associated value i.e `urlMap: {artifact_id: 'route_stage'}` so used index[0].

And this is my first contribution in pipelines code, in order to fix this issue, can see any associated spec files else would have updated the tests with this scenario.